src/controllers/services.ts

```diff
@@ -192,3 +211,49 @@ export async function putService(
     ResponseSuccessJson
   );
 }
+
+/**
+ * Upload service logo for/with a specific serviceId.
+ */
+export async function putServiceLogo(
+  apiClient: ApiManagementClient,
+  authenticatedUser: AdUser,
+  serviceId: NonEmptyString,
+  serviceLogo: ApiLogo
+): Promise<IResponseSuccessRedirectToResource<{}, {}> | ErrorResponses> {
+  return getApimUserTask(apiClient, authenticatedUser)
+    .chain(user =>
+      checkAdminTask(user).chain(() =>
+        uploadServiceLogoTask(serviceId, serviceLogo).map(result => result)
+      )
+    )
+    .fold<IResponseSuccessRedirectToResource<{}, {}> | ErrorResponses>(
+      identity,
+      identity
+    )
+    .run();
+}
+
+/**
+ * Upload organization logo for/with a specific serviceId.
+ */
+export async function putOrganizationLogo(
+  apiClient: ApiManagementClient,
+  authenticatedUser: AdUser,
+  organizationFiscalCode: NonEmptyString,
+  serviceLogo: ApiLogo
+): Promise<IResponseSuccessRedirectToResource<{}, {}> | ErrorResponses> {
+  return getApimUserTask(apiClient, authenticatedUser)
+    .chain(user =>
+      checkAdminTask(user).chain(() =>
+        uploadOrganizationLogoTask(organizationFiscalCode, serviceLogo).map(
+          result => result
+        )
+      )
```

Comment:
Can we flat this chain's pipeline ? i.e:
```suggestion
    .chain(user =>
      checkAdminTask(user)
     ).chain(() =>
        uploadOrganizationLogoTask(organizationFiscalCode, serviceLogo)
     )
```